packages/flutter/lib/src/material/dialog.dart

```diff
@@ -1308,6 +1387,58 @@ Future<T?> showDialog<T>({
   ));
 }
 
+/// Displays either a Material or Cupertino dialog depending on platform.
+///
+/// On most platforms this function will act the same as [showDialog], except
+/// for iOS and macOS, in which case it will act the same as
+/// [showCupertinoDialog].
+///
+/// On Cupertino platforms, [barrierColor], [useSafeArea], and
+/// [traversalEdgeBehavior] are ignored.
+Future<T?> showAdaptiveDialog<T>({
```

Comment:
Do we have a pattern of adaptive functions? I wasn't sure the right way to name this.

Reply:
@HansMuller if you could take a quick peek at this.